src/execution.ts

```diff
@@ -1493,7 +1489,7 @@ function defaultResolveTypeFn(
           `Promises are not supported for resolving type of ${value}`
         );
       } else if (isTypeOfResult) {
-        return type;
+        return type.name;;
```

Comment:
Can you justify this change here?

Reply:
In GraphQL v16, type resolvers cannot return type itself but only the name of the type.